sdks/agent-sdk/src/core/Agent.ts

```diff
@@ -0,0 +1,132 @@
+import type { ContentCodec } from "@xmtp/content-type-primitives";
+import {
+  Client,
+  type ClientOptions,
+  type DecodedMessage,
+  type Identifier,
+  type Signer,
+} from "@xmtp/node-sdk";
+import { type MessageFilter } from "@/filters/MessageFilters";
+import { AgentContext } from "./AgentContext";
+import { AgentEventEmitter } from "./AgentEventEmitter";
+
+export type AgentMiddleware<ContentTypes> = (
+  ctx: AgentContext<ContentTypes>,
+  next: () => Promise<void>,
+) => Promise<void>;
+
+type AgentConfig<ContentCodecs extends ContentCodec[] = []> =
+  | {
+      signer: Signer;
+      options?: Omit<ClientOptions, "codecs"> & { codecs?: ContentCodecs };
+    }
+  | {
+      identifier: Identifier;
+      options?: Omit<ClientOptions, "codecs"> & { codecs?: ContentCodecs };
+    };
+
+export type MessageHandler = {
+  filter?: MessageFilter;
+  handler: MessageHandler;
+};
+
+/**
+ * XMTP Agent for handling messages and events.
+ */
+export class Agent<ContentTypes> extends AgentEventEmitter<ContentTypes> {
+  private client: Client<ContentTypes>;
+  private middleware: AgentMiddleware<ContentTypes>[] = [];
+  private isListening = false;
+
+  constructor(client: Client<ContentTypes>) {
+    super();
+    this.client = client;
+  }
+
+  static async create<ContentCodecs extends ContentCodec[] = []>(
+    config: AgentConfig<ContentCodecs>,
+  ) {
+    const client =
+      "signer" in config
+        ? await Client.create(config.signer, config.options)
+        : await Client.build(config.identifier, config.options);
+    return new Agent(client);
+  }
```

Comment:
WDYT of having `Agent.create` and `Agent.build` to make it clear that it aligns with `Client.create` and `Client.build`?

Reply:
Agreed. I added `build` and `create`.